### PR TITLE
feat: add preStart and postStop scripts

### LINKED ISSUE #262: Add preStop/postStart lifecycle hooks for managing maintenance mode
There are preStop and postStart scripts in the operator: https://github.com/redpanda-data/redpanda/blob/e51d5b7f2ef76d5160ca01b8c7a8cf07593d29b6/src/go/k8s/pkg/resources/secret.go

Duplicate that functionality in the helm chart.

It would involve generating the scripts in a secret, then mounting them in the sts and linking them to the sts [lifecycle](https://github.com/redpanda-data/redpanda/blob/54bccf3a29b173ef50134fbc128985cc7cf1fd11/src/go/k8s/pkg/resources/statefulset.go#L569).

### PR DESCRIPTION
**Pull request by alejandroEsc**
Work in progress:

Adding postStart and preStop scripts as found on the operator code. 

Fixes #262

### CODE REVIEW

File: charts/redpanda/templates/statefulset.yaml [hunk 1]
```diff
@@ -164,6 +163,19 @@ spec:
...
+          lifecycle:
+            postStart:
+              exec:
+                command:
+                  - "/bin/sh"
+                  - "-c"
+                  - "./var/lifecycle/postStart.sh"
+            preStop:
+              exec:
+                command:
```
Comment (joejulian) [inline charts/redpanda/templates/statefulset.yaml:175]: `sh` here, but `bash` in the scripts themselves. Let's just use bash for both.

File: charts/redpanda/templates/statefulset.yaml [hunk 2]
```diff
@@ -164,6 +163,19 @@ spec:
...
+                  - "/bin/sh"
+                  - "-c"
+                  - "./var/lifecycle/preStop.sh"
```
Comment (joejulian) [inline charts/redpanda/templates/statefulset.yaml:178]: `sh` here, but `bash` in the scripts themselves. Let's just use bash for both.

**Review by joejulian — COMMENTED** [review]
Looking good! Now we just have to figure out how to test this.

**Comment by alejandroEsc** [comment]
> Looking good! Now we just have to figure out how to test this.

We could just try to run each command on one of the machines (this checks that they exist and that they run without error). The other tests that bring up and down a cluster will be enough to tell us if there is an issue with the scripts because any failure in the scripts will show up as unable to bring up a redpanda cluster or take down a redpanda cluster. Since the scripts depend on TLS and TLS related options they will definitely fail with broken changes, bad options, or unexpected changes to the logic.

**Comment by joejulian** [comment]
> > Looking good! Now we just have to figure out how to test this.
> 
> We could just try to run each command on one of the machines (this checks that they exist and that they run without error). The other tests that bring up and down a cluster will be enough to tell us if there is an issue with the scripts because any failure in the scripts will show up as unable to bring up a redpanda cluster or take down a redpanda cluster. Since the scripts depend on TLS and TLS related options they will definitely fail with broken changes, bad options, or unexpected changes to the logic.

Yeah, that's probably good enough for now.

**Review by joejulian — APPROVED** [review]

File: charts/redpanda/templates/_helpers.tpl [hunk 3]
```diff
@@ -460,3 +460,23 @@ fsGroup: {{ dig "podSecurityContext" "fsGroup" .Values.statefulset.securityConte
 runAsUser: {{ dig "podSecurityContext" "runAsUser" .Values.statefulset.securityContext.runAsUser .Values.statefulset }}
 runAsGroup: {{ dig "podSecurityContext" "fsGroup" .Values.statefulset.securityContext.fsGroup .Values.statefulset }}
 {{- end -}}
+
+{{- define "tls-curl-flags" -}}
+  {{- $result := "" -}}
+  {{- if (include "tls-enabled" . | fromJson).bool -}}
+    {{- $path := (printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert) -}}
+    {{- $result = (printf "--cacert %s/tls.crt" $path) -}}
```
Comment (jan-g) [inline charts/redpanda/templates/_helpers.tpl:468]: Noting for posterity - this looks like a C&P error. Should be `ca.crt` here.

**Pull request merged**